scripts/airgapped/deploy-1.9.sh

```diff
@@ -0,0 +1,182 @@
+    --config custom_images="visualization_server: '$(img visualization-server)'
+frontend: '$(img frontend)'
+"
+
+juju deploy --trust --debug ./$(charm kfp-schedwf) --resource oci-image=$(img scheduledworkflow)
+juju deploy --trust --debug ./$(charm kfp-ui) --resource ml-pipeline-ui=$(img frontend)
+juju deploy --trust --debug ./$(charm kfp-viewer) --resource kfp-viewer-image=$(img viewer-crd-controller)
+juju deploy --trust --debug ./$(charm kfp-viz) --resource oci-image=$(img visualization-server)
+juju deploy --trust --debug ./$(charm knative-eventing) --config namespace=knative-eventing \
+     --config custom_images="eventing-controller/eventing-controller: $(img knative-releases/knative.dev/eventing/cmd/controller)
+eventing-webhook/eventing-webhook: $(img knative-releases/knative.dev/eventing/cmd/webhook)
+imc-dispatcher/dispatcher: $(img knative-releases/knative.dev/eventing/cmd/in_memory/channel_dispatcher)
+imc-controller/controller: $(img knative-releases/knative.dev/eventing/cmd/in_memory/channel_controller)
+mt-broker-controller/mt-broker-controller: $(img knative-releases/knative.dev/eventing/cmd/mtchannel_broker)
+mt-broker-filter/filter: $(img knative-releases/knative.dev/eventing/cmd/broker/filter)
+mt-broker-ingress/ingress: $(img knative-releases/knative.dev/eventing/cmd/broker/ingress)
+pingsource-mt-adapter/dispatcher: $(img knative-releases/knative.dev/eventing/cmd/mtping)
+"
+juju deploy --trust --debug ./$(charm knative-operator) --resource knative-operator-image=$(img knative-releases/knative.dev/operator/cmd/operator) --resource knative-operator-webhook-image=$(img knative-releases/knative.dev/operator/cmd/webhook) --config otel-collector-image=$(img otel/opentelemetry-collector)
+juju deploy --trust --debug ./$(charm knative-serving) --config namespace=knative-serving --config istio.gateway.namespace=kubeflow --config istio.gateway.name=kubeflow-gateway \
+    --config queue_sidecar_image="$(img knative-releases/knative.dev/serving/cmd/queue)" \
+    --config custom_images="activator: $(img knative-releases/knative.dev/serving/cmd/activator)
+autoscaler: $(img knative-releases/knative.dev/serving/cmd/autoscaler)
+controller: $(img knative-releases/knative.dev/serving/cmd/controller)
+webhook: $(img knative-releases/knative.dev/serving/cmd/webhook)
+autoscaler-hpa: $(img knative-releases/knative.dev/serving/cmd/autoscaler-hpa)
+net-istio-controller/controller: $(img knative-releases/knative.dev/net-istio/cmd/controller)
+net-istio-webhook/webhook: $(img knative-releases/knative.dev/net-istio/cmd/webhook)
+queue-proxy: $(img knative-releases/knative.dev/serving/cmd/queue)
+domain-mapping: $(img knative-releases/knative.dev/serving/cmd/domain-mapping:)
+domainmapping-webhook: $(img knative-releases/knative.dev/serving/cmd/domain-mapping-webhook)
+migrate: $(img knative-releases/knative.dev/pkg/apiextensions/storageversion/cmd/migrate)
+"
```

Comment:
during testing, the deployment command I got was:
```
13:26:47 DEBUG juju.cmd supercommand.go:57   args: []string{"/snap/juju/28056/bin/juju", "deploy", "--trust", "--debug", "./knative-serving_57d9e75.charm", "--config", "namespace=knative-serving", "--config", "istio.gateway.namespace=kubeflow", "--config", "istio.gateway.name=kubeflow-gateway", "--config", "queue_sidecar_image=172.17.0.2:5000/knative-releases/knative.dev/serving/cmd/queue:89e6f90141f1b63405883fbb4de0d3b6d80f8b77e530904c4d29bdcd1dc5a167", "--config", "custom_images=activator: 172.17.0.2:5000/knative-releases/knative.dev/serving/cmd/activator:ad42ddc9bc4e25fdc88c240d7cbfad4b2708eb7d26e07ae904d258011141116e\nautoscaler: 172.17.0.2:5000/knative-releases/knative.dev/serving/cmd/autoscaler:66aa0dbceee62691d5327e423bbd7cbd411903747adeab61fdc81b14590793d4\ncontroller: 172.17.0.2:5000/knative-releases/knative.dev/serving/cmd/controller:e5b7b6edd265b66d32f424bd245c06455154462ade6ce05698472212248d5657\nwebhook: 172.17.0.2:5000/knative-releases/knative.dev/serving/cmd/webhook:48aee2733721ecc77956abc5a2ca072853a669ebc97519beb48f7b3da8455e67\nautoscaler-hpa: 172.17.0.2:5000/knative-releases/knative.dev/serving/cmd/autoscaler-hpa:f5d84faf112df9a63c5339a2843f6fcce6044c71541e9e1989e922885e927922\nnet-istio-controller/controller: \nnet-istio-webhook/webhook: \nqueue-proxy: 172.17.0.2:5000/knative-releases/knative.dev/serving/cmd/queue:89e6f90141f1b63405883fbb4de0d3b6d80f8b77e530904c4d29bdcd1dc5a167\ndomain-mapping: \ndomainmapping-webhook: \nmigrate: 172.17.0.2:5000/knative-releases/knative.dev/pkg/apiextensions/storageversion/cmd/migrate:b557a57aeef5c530871b59bfc6d2fdfea54e34858c5b2186b1d501bf8133d5ff\n"}
```
the following images are missing from the `custom_images`:
* net-istio-controller/controller
* net-istio-webhook/webhook
* domain-mapping
* domainmapping-webhook

Reply:
[This PR](https://github.com/canonical/knative-operators/pull/221) Has fixed to issues for `net-istio-controller/controller` and `net-istio-webhook/webhook`, working on the other 2

Reply:
It seems that the latest version of `knative` doesn't include those images in the manifests, so I updated the script to reflect that. I'll also create an issue to remove them in the charm's repo

Reply:
agree for the domain mapping images

As for the `net-istio-*` images, it seems that https://github.com/canonical/knative-operators/pull/221 was not cherry-picked to `track/1.12`. I still don't see it in the `retagged-images.txt`.

Reply:
I created [the PR](https://github.com/canonical/knative-operators/pull/222) for it!